x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/store/input/DirectBlobContainerIndexInput.java

```diff
@@ -291,23 +338,11 @@ public String toString() {
     }
 
     private InputStream openBlobStream(int part, long pos, long length) throws IOException {
-        assert assertCurrentThreadMayAccessBlobStore();
+        assert MetadataCachingIndexInput.assertCurrentThreadMayAccessBlobStore();
         stats.addBlobStoreBytesRequested(length);
         return blobContainer.readBlob(fileInfo.partName(part), pos, length);
     }
 
-    /**
-     * Fully read up to {@code length} bytes from the given {@link InputStream}
-     */
-    private static int readFully(InputStream inputStream, final ByteBuffer b, int length, CheckedRunnable<IOException> onEOF)
```

Comment:
This one I inlined in the two use cases it had since it was really weird and I wanted to see if I can spot some more opportunities for drying up the physical reading of things.